### PR TITLE
Add AdvertisementEntry.spec.tsx for Vitest migration

### PR DESCRIPTION
**Pull request by IITI-tushar**
**What kind of change does this PR introduce?**

refactoring

**Issue Number:** #2484 

Fixes ##2484

**Did you add tests for your changes?**

Vitest tes passed :
image

coverage for AdvertisementEntry.spec.tsx file
image

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**
Replace Jest-specific functions and mocks with Vitest equivalents and rename from test.tsx to spec.tsx

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

yes

## Summary by CodeRabbit

- **Tests**
	- Introduced a comprehensive suite of tests for the AdvertisementEntry component, covering rendering, updating, and deleting advertisements, as well as handling default props and error scenarios.
	- Enhanced testing for dropdown functionality and mutation scenarios, ensuring robust coverage of the component's behavior.

### CODE REVIEW

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx
Comment (IITI-tushar) [inline src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx:1]: I have only made migration from jest to vitest for 
" src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.test.tsx ".

**Comment by palisadoes** [comment]
Please fix:

1. Any failing tests
1. Any conflicting files

Make sure CodeRabbit.ai approves your changes

**Comment by palisadoes** [comment]
Please fix:

1. Any failing tests (Click on the `details` link for more information)
1. Any conflicting files

Make sure CodeRabbit.ai approves your changes

**Comment by IITI-tushar** [comment]
@palisadoes I have made changes according to AdvertisementEntry.test.tsx , and coderabbitbot is asking me to make changes which conflict the AdvertisementEntry.test.tsx , So should I go with the changes coderabbit is suggesting , or just resolve the conversation?
Eg: 
This is from advertisementEntry.test.tsx 
image
code rabbit is suggesting this:
image

**Comment by Cioppolo14** [comment]
@IITI-tushar Please address the coderabbit comments and fix the failed tests.

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+  //   expect(updateAdByIdMock).toHaveBeenCalledWith({
+  //     variables: {
+  //       id: '1',
+  //       name: 'Updated Advertisement',
+  //       type: 'BANNER',
+  //       startDate: dayjs().add(1, 'day').format('YYYY-MM-DD'),
+  //       endDate: dayjs().add(2, 'days').format('YYYY-MM-DD'),
+  //     },
+  //   });
+  // });
```
[the remark this answers is not in the document]
Comment (IITI-tushar) [reply src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx:390]: coderabbit asked to add this case, but it was giving errors and this case was not found in AdvertisementEntry.test.tsx from which Advertisement.spec.tsx got migrated

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+      });
+      const deletionFailedText = screen.queryByText((content, element) => {
+        return (
+          element?.textContent === 'Deletion Failed' &&
+          element.tagName.toLowerCase() === 'div'
+        );
+      });
+      expect(deletionFailedText).toBeNull();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (IITI-tushar) [reply src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx:712]: this case was in AdvertisementEntry.test.tsx from which Advertisement.spec.tsx got migrated.
 so not deleting it.

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+    expect(updateAdByIdMock).toHaveBeenCalledWith({
+      variables: {
+        id: '1',
+        name: 'Updated Advertisement',
+        type: 'BANNER',
+        startDate: dayjs().add(1, 'day').format('YYYY-MM-DD'),
+        endDate: dayjs().add(2, 'days').format('YYYY-MM-DD'),
+      },
+    });
+  });
```
[the remark this answers is not in the document]
Comment (IITI-tushar) [reply src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx:326]: this case was not found in AdvertisementEntry.test.tsx from which Advertisement.spec.tsx got migrated.

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+    const mediaElement = screen.getByTestId('media');
+    expect(mediaElement).toHaveAttribute('src', '');
+
+    // Check that the component renders with default `endDate`
+    const defaultEndDate = new Date().toDateString();
+    expect(screen.getByText(`Ends on ${defaultEndDate}`)).toBeInTheDocument();
+
+    // Check that the component renders with default `startDate`
+    const defaultStartDate = new Date().toDateString();
+    expect(screen.getByText(`Ends on ${defaultStartDate}`)).toBeInTheDocument(); //fix text "Ends on"?
```
[the remark this answers is not in the document]
Comment (IITI-tushar) [reply src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx:168]: this was not found in AdvertisementEntry.test.tsx from which Advertisement.spec.tsx got migrated
 and replacing ends with starts gives error.

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+        : void {
+          throw new Error('Function not implemented.');
+        }}
+      />,
+    );
+
+    // Check that the component renders with provided values
+    expect(getByText(mockName)).toBeInTheDocument();
+    // Add more checks based on how each prop affects rendering
+  });
```
[the remark this answers is not in the document]
Comment (IITI-tushar) [reply src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.spec.tsx]: added

**Comment by IITI-tushar** [comment]
@Cioppolo14 @palisadoes I addressed code rabbit comments. and all tests are passed

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please remove the commented out code in the test file.

**Comment by IITI-tushar** [comment]
> 1. Please remove the commented out code in the test file.
> 2. The tests were minimal before. You've improved the code quality. Thanks.
@palisadoes 
Done 👍

**Review by palisadoes — APPROVED** [review]

**Pull request merged**